src/cosim/proxy/remote_slave.cpp

```diff
@@ -0,0 +1,197 @@
+    auto status = slave_->get_boolean(vr, data);
+    if (!status) {
+        COSIM_PANIC();
+    }
+    for (unsigned int i = 0; i < data.size(); i++) {
+        values[i] = data[i];
+    }
+}
+
+void cosim::proxy::remote_slave::get_string_variables(gsl::span<const cosim::value_reference> variables,
+    gsl::span<std::string> values) const
+{
+    COSIM_INPUT_CHECK(variables.size() == values.size());
+    if (variables.empty()) return;
+
+    const auto vr = std::vector<proxyfmu::fmi::value_ref>(variables.begin(), variables.end());
+    auto data = std::vector<std::string>(vr.size());
+    auto status = slave_->get_string(vr, data);
+    if (!status) {
+        COSIM_PANIC();
+    }
+    for (unsigned int i = 0; i < data.size(); i++) {
+        values[i] = data[i];
+    }
+}
+
+void cosim::proxy::remote_slave::set_real_variables(gsl::span<const cosim::value_reference> variables,
+    gsl::span<const double> values)
+{
+    COSIM_INPUT_CHECK(variables.size() == values.size());
+    if (variables.empty()) return;
+
+    const auto vr = std::vector<proxyfmu::fmi::value_ref>(variables.begin(), variables.end());
+    const auto _values = std::vector<double>(values.begin(), values.end());
+    auto status = slave_->set_real(vr, _values);
+    if (!status) {
+        COSIM_PANIC();
+    }
+}
+
+void cosim::proxy::remote_slave::set_integer_variables(gsl::span<const cosim::value_reference> variables,
+    gsl::span<const int> values)
+{
+    COSIM_INPUT_CHECK(variables.size() == values.size());
+    if (variables.empty()) return;
+
+    const auto vr = std::vector<proxyfmu::fmi::value_ref>(variables.begin(), variables.end());
+    const auto _values = std::vector<int>(values.begin(), values.end());
+    auto status = slave_->set_integer(vr, _values);
+    if (!status) {
+        COSIM_PANIC();
+    }
+}
+
+void cosim::proxy::remote_slave::set_boolean_variables(gsl::span<const cosim::value_reference> variables,
+    gsl::span<const bool> values)
+{
+    COSIM_INPUT_CHECK(variables.size() == values.size());
+    if (variables.empty()) return;
+
+    const auto vr = std::vector<proxyfmu::fmi::value_ref>(variables.begin(), variables.end());
+    const auto _values = std::vector<bool>(values.begin(), values.end());
+    auto status = slave_->set_boolean(vr, _values);
+    if (!status) {
+        COSIM_PANIC();
+    }
+}
+
+void cosim::proxy::remote_slave::set_string_variables(gsl::span<const cosim::value_reference> variables,
+    gsl::span<const std::string> values)
+{
+    COSIM_INPUT_CHECK(variables.size() == values.size());
+    if (variables.empty()) return;
+
+    const auto vr = std::vector<proxyfmu::fmi::value_ref>(variables.begin(), variables.end());
+    const auto _values = std::vector<std::string>(values.begin(), values.end());
+    auto status = slave_->set_string(vr, _values);
+    if (!status) {
+        COSIM_PANIC();
+    }
+}
+
+cosim::proxy::remote_slave::~remote_slave()
+{
+    remote_slave::end_simulation();
```

Comment:
I haven't looked into the functions you're calling here, but I just want to note that one cannot throw exceptions from destructors(*). So if there is a non-neglectible chancethat these functions may throw, you may want to wrap the calls in a `try`-`catch`-statement and log a message on error.

(*) Or, you can, but it causes the program to immediately crash via `std::terminate()`.